### DISCUSSION TITLE
Unable to get owner error on uploading file using supabasejs

### DISCUSSION THREAD

**Discussion by MicrorealHQ** [supabase/supabase#23638] [Questions]
I am trying to upload a file to storage using supabase-js but I got error "Unable to get owner", here is full log:

> `error": [
>       {
>         "message": "invalid signature",
>         "name": "Error",
>         "raw": "{\"metadata\":{},\"code\":\"AccessDenied\",\"httpStatusCode\":403,\"userStatusCode\":400,\"originalError\":{\"name\":\"JsonWebTokenError\",\"message\":\"invalid signature\"},\"error\":\"Unauthorized\"}",
>         "stack": "Error: invalid signature\n    at Object.AccessDenied (/app/dist/storage/errors.js:86:34)\n    at /app/dist/auth/jwt.js:110:48\n    at /app/node_modules/jsonwebtoken/verify.js:171:14\n    at /app/dist/auth/jwt.js:82:9\n    at module.exports [as verify] (/app/node_modules/jsonwebtoken/verify.js:101:10)\n    at /app/dist/auth/jwt.js:108:32\n    at new Promise (<anonymous>)\n    at verifyJWT (/app/dist/auth/jwt.js:107:12)\n    at Object.<anonymous> (/app/dist/http/plugins/jwt.js:19:56)"
>       }
>     ],`

The token that was exchanged and used in supabase client is using for other actions (select, insert) in other tables normally.
Here is payload of the exchanged auth token:
`const payload = {
        aud: "authenticated",
        iat,
        exp: iat + 3 * 24 * 60 * 60,
        sub: userId,
        role: "authenticated",
      };`
But when we try to use the same supabase client to upload file, this error occur.
I think it's related to how we setup policies for storage:
`CREATE POLICY "allow user upload file"
    ON storage.objects
    FOR INSERT
    TO authenticated
    WITH CHECK (
            bucket_id = 'media'
    );`
It's quite strange because of this policy setup since I am able to upload from flutter project using the same exchange token
So I couldn't find any suspect here.
Is there anyone here got this issue or any suggestion to solve this?

**Comment by silentworks** [comment]
Does it work when RLS is turned off on the storage object or if you set the `INSERT` policy to just `true`? Is your user authenticated?

**Reply by MicrorealHQ** [reply]
I didn't want to change this yet since the flutter project are working fine. Is there any case this issue related to supabasejs library? I am using v2.26.0. Let me try to upgrade supabase version

**Reply by MicrorealHQ** [reply]
I still got that error on the latest version as well

**Comment by MicrorealHQ** [comment]
<html>
<body>

InvalidSignature | The signature provided does not match the calculated signature. | 403 | Check that you are providing the correct signature format, for more information refer to SignatureV4
-- | -- | -- | --



</body>
</html>
Seems like the exchange token works for other database action InvalidSignature	The signature provided does not match the calculated signature.	403	Check that you are providing the correct signature format, for more information refer to [SignatureV4](https://docs.aws.amazon.com/AmazonS3/latest/API/sig-v4-authenticating-requests.html)

https://supabase.com/docs/guides/storage/debugging/error-codes

As far as I understand, this is the error I got due to difference of signature format. But it doesn't make any sense that the flutter project using the same exchange token and can upload normally

**Comment by Ronneback** [comment]
+1 anyone having a solution on this InvalidSignature 403?

**Comment by mashanz** [comment]
got then same issue no fix yet
